src/Core/Grid/Definition/Factory/SearchEngineGridDefinitionFactory.php

```diff
@@ -0,0 +1,225 @@
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Definition\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Action\Bulk\BulkActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Action\Bulk\BulkActionCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Action\GridActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Action\GridActionCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\RowActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\RowActionCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\Type\LinkRowAction;
+use PrestaShop\PrestaShop\Core\Grid\Action\Type\SimpleGridAction;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\BulkActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\DataColumn;
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollectionInterface;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+/**
+ * Class SearchEngineGridDefinitionFactory creates definition for search engines grid.
+ */
+final class SearchEngineGridDefinitionFactory extends AbstractFilterableGridDefinitionFactory
+{
+    /**
+     * @var string
+     */
+    public const GRID_ID = 'search_engine';
+
+    use BulkDeleteActionTrait;
+    use DeleteActionTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId(): string
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName(): string
+    {
+        return $this->trans('Search Engines', [], 'Admin.Navigation.Menu');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns(): ColumnCollectionInterface
+    {
+        return (new ColumnCollection())
+            ->add(
+                (new BulkActionColumn('bulk'))
+                    ->setOptions([
+                        'bulk_field' => 'id_search_engine',
+                    ])
+            )
+            ->add(
+                (new DataColumn('id_search_engine'))
+                    ->setName($this->trans('ID', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'id_search_engine',
+                    ])
+            )
+            ->add(
+                (new DataColumn('server'))
+                    ->setName($this->trans('Server', [], 'Admin.Shopparameters.Feature'))
+                    ->setOptions([
+                        'field' => 'server',
+                    ])
+            )
+            ->add(
+                (new DataColumn('getvar'))
```

Comment:
I wonder if it's not the right moment to rename - in the code only, not in the interface - this weird parameter "$_GET variable".

This name does sound very good to me 😄 could we find a better name ? Something that helps the merchant or the developer to understand quickly what use it has?

Reply:
I have no idea what this name should be.

Reply:
maybe `query key`?

Reply:
So in code, I should rename all `getvar`s to `query_key`s?